### ISSUE TITLE
Allow overwriting default selector for focus trap

### ISSUE THREAD

**Issue by blaky** [Sage/carbon#5555]
### Desired behaviour

To be able to set the selector that is used by the focus trap to identify the focusable elements.

### Current behaviour

The consuming project cannot control what elements considered to be focusable by the focus trap of Carbon, the selector is hardcoded in src/__internal__/focus-trap/focus-trap-utils.ts:1 as `defaultFocusableSelectors`, but there is no way to override this default.
In our project we have some cases when the input element is rendered into a hidden container, which results focus loss. Sometimes when tabbing out from a date field, the next focusable element is identified as the date picker, which is closed immediately after it gets focused, I couldn't reproduce this in isolation, probably it only happens if you have a large set of event handlers. 

### Suggested Solution

Add a new `focusTrapSelector` prop to dialogs and the sidebar component so the consumer projects can control what should be considered as focusable. 

It should be just the case of exposing this prop on the dialog-like components  and pass it forward to the focus trap. If the prop is not provided the current `defaultFocusableSelectors` selector value would be used.

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

**Comment by edleeks87** [comment]
https://codesandbox.io/s/carbon-quickstart-forked-3ludjo?file=/src/index.js Sandbox demo of similar issue where input in a hidden container is included focusable elements.

I don't think it's possible for us to write a default query string that is general enough to catch all potential uses cases, so adding the ability for consumers to provide their own override seems like a good approach

**Comment by robinzigmond** [comment]
FE-5495

**Comment by DlgSHi** [comment]
released in [PR5557](https://github.com/Sage/carbon/pull/5557)
